dolphin-dnn/src/main/java/edu/snu/dolphin/dnn/conf/NeuralNetworkConfigurationBuilder.java

```diff
@@ -35,25 +37,18 @@
  */
 public final class NeuralNetworkConfigurationBuilder implements Builder<Configuration> {
 
-  private Collection<String> layerConfigurations = new ArrayList<>();
-  private int index = 0;
+  private ArrayList<Configuration> layerConfigurations = new ArrayList<>();
```

Comment:
This variable is guaranteed to have O(1) `get()` performance even if you declare it as a `List`, because you're initializing it with `new ArrayList<>()`. We should follow the convention of using interfaces instead of concrete classes for declaring variables, when we can.